### DISCUSSION TITLE
Viewing usernames and emails who make commits in an organisation

### DISCUSSION THREAD

**Discussion by TomDougall** [community/community#46751] [Enterprise]
Hi, I am trying to see whether it is possible to see usernames and/or emails of people who makes commits even when they've used no-reply or hidden their emails in their profile,

It would be important for an audit log, is this something that would only be available from a self-managed enterprise solution?

What options are there for ensuring there is a proper audit log, while also allowing users to hide their email on public repos in an organisation.

Thanks

**Comment by chris-c-thomas** [comment]
Not sure I agree with auditing people but you can use the API to grab the commit history of the repository and see everything. Even with the hide my email feature the username will still appear in the log. It's just to hide the email portion. You can't really hide commits from the git log without deleting them.
